### PR TITLE
Fix android RTL issues

### PR DESCRIPTION
**Pull request by yuvalsho**

### CODE REVIEW

File: src/timeline-list/index.tsx [hunk 1]
```diff
@@ -73,6 +75,12 @@ const TimelineList = (props: TimelineListProps) => {
     prevDate.current = date;
   }, [updateSource]);
 
+  const shouldApplyRTL = useMemo(() => constants.isRTL && constants.isAndroid, []);
```
Comment (Inbal-Tish) [inline src/timeline-list/index.tsx:78]:
```suggestion
  const shouldUseAndroidRTLFix = useMemo(() => constants.isRTL && constants.isAndroid, []);
```
Just to keep unified naming, or better let's move this to the 'commons/constants' file

File: src/expandableCalendar/WeekCalendar/index.tsx [hunk 2]
```diff
@@ -70,14 +70,15 @@ const WeekCalendar = (props: WeekCalendarProps) => {
           }) :
           sameWeek(item, date, firstDay));
       if (pageIndex !== currentIndex.current) {
+        const adjustedIndexFrScroll = APPLY_ANDROID_FIX ? NUM_OF_ITEMS - 1 - pageIndex : pageIndex;
```
Comment (Inbal-Tish) [inline src/expandableCalendar/WeekCalendar/index.tsx:73]: 'shouldUseAndroidRTLFix' should be the name, and we need to take it from the constants

File: src/calendar-list/index.tsx [hunk 3]
```diff
@@ -108,13 +109,16 @@ const CalendarList = (props: CalendarListProps & ContextProp, ref: any) => {
 
   const [currentMonth, setCurrentMonth] = useState(parseDate(current));
 
+  const isAndroidRTL = useMemo(() =>constants.isRTL && constants.isAndroid && horizontal, [horizontal]);
```
Comment (Inbal-Tish) [inline src/calendar-list/index.tsx:112]:
```suggestion
  const isAndroidRTL = useMemo(() => shouldUseAndroidRTLFix && horizontal, [horizontal]);
```
Need to take 'shouldUseAndroidRTLFix' from commons/constants

File: src/calendar-list/index.tsx [hunk 4]
```diff
@@ -99,7 +100,7 @@ const CalendarList = (props: CalendarListProps & ContextProp, ref: any) => {
     onMomentumScrollEnd,
     /** FlatList props */
     onEndReachedThreshold,
-    onEndReached
+    onEndReached,
```
Comment (Inbal-Tish) [inline src/calendar-list/index.tsx:103]: no trailing commas please

File: src/infinite-list/index.tsx [hunk 5]
```diff
@@ -68,6 +70,9 @@ const InfiniteList = (props: InfiniteListProps, ref: any) => {
       }
     )
   );
+  const shouldUseAndroidRTLFix = useMemo(() => {
+    return constants.isAndroid && constants.isRTL && isHorizontal;
```
Comment (Inbal-Tish) [inline src/infinite-list/index.tsx:74]:
```suggestion
    return constants.isAndroidRTL && isHorizontal;
```

File: src/timeline-list/useTimelinePages.ts [hunk 6]
```diff
@@ -26,6 +26,9 @@ const UseTimelinePages = ({date, listRef, numberOfDays}: UseTimelinePagesProps)
 
   const [pages, setPages] = useState<string[]>(pagesRef.current);
   const shouldResetPages = useRef(false);
+  const shouldUseAndroidRTLFix = useMemo(() => {
+    return constants.isAndroid && constants.isRTL;
```
Comment (Inbal-Tish) [inline src/timeline-list/useTimelinePages.ts:30]:
```suggestion
    return constants.isAndroidRTL;
```

File: src/calendar-list/index.tsx [hunk 7]
```diff
@@ -250,11 +254,21 @@ const CalendarList = (props: CalendarListProps & ContextProp, ref: any) => {
...
+  const onPressArrowRight = useCallback((method, month) => {
+    headerProps.onPressArrowRight?.(method, month);
+  }, [headerProps.onPressArrowRight]);
+
   const renderStaticHeader = () => {
     if (staticHeader && horizontal) {
       return (
         <CalendarHeader
           {...headerProps}
+          onPressArrowLeft={headerProps.onPressArrowLeft ? onPressArrowLeft : undefined}
```
Comment (Inbal-Tish) [inline src/calendar-list/index.tsx:270]: Why do we need to implement these callbacks? The CalendarHeader is invoking them by default..

File: src/infinite-list/index.tsx [hunk 8]
```diff
@@ -82,7 +87,7 @@ const InfiniteList = (props: InfiniteListProps, ref: any) => {
     }
 
     setTimeout(() => {
-      const x = isHorizontal ? Math.floor(data.length / 2) * pageWidth : 0;
+      const x = isHorizontal ? constants.isRTL && constants.isAndroid ? Math.floor(data.length / 2) + 1 : Math.floor(data.length / 2) * pageWidth : 0;
```
Comment (Inbal-Tish) [inline src/infinite-list/index.tsx:90]:
```suggestion
      const x = isHorizontal ? constants.isAndroidRTL ? Math.floor(data.length / 2) + 1 : Math.floor(data.length / 2) * pageWidth : 0;
```

File: src/calendar-list/index.tsx (hunk 7 above)
Comment (yuvalsho) [reply src/calendar-list/index.tsx:270]: you are right, they were part of a previous attempt to fix and are obsolete now

File: src/calendar-list/index.tsx [hunk 9]
```diff
@@ -304,6 +302,7 @@ const CalendarList = (props: CalendarListProps & ContextProp, ref: any) => {
       <FlatList
         // @ts-expect-error
         ref={list}
+        windowSize={constants.isAndroidRTL && horizontal ? pastScrollRange + futureScrollRange + 1 : undefined}
```
Comment (Inbal-Tish) [inline src/calendar-list/index.tsx:305]: We should use 'shouldUseAndroidRTLFix' here

**Review by Inbal-Tish — APPROVED** [review]

**Pull request merged**